src/shared/blockdev-util.c

```diff
@@ -418,6 +418,13 @@ int blockdev_partscan_enabled(int fd) {
         if (r != -ENOENT)
                 return r;
 
+        /* Partition block devices never have partition scanning on, there's no concept of sub-partitions for
+         * partitions. */
+        const char *devtype;
```

Comment:
This needs to be initialized since we ignore the error from function call?

Or, use if (...get_devtype() >= 0 && streq(...)) below?